src/rapidgzip/gzip/GzipReader.hpp

```diff
@@ -495,7 +646,7 @@ GzipReader::readGzipFooter()
 {
     const auto footer = rapidgzip::gzip::readFooter( m_bitReader );
 
-    if ( static_cast<uint32_t>( m_streamBytesCount ) != footer.uncompressedSize ) {
+    if ( m_didReadHeader && static_cast<uint32_t>( m_streamBytesCount ) != footer.uncompressedSize ) {
```

Comment:
A simple `m_blockMap` (!= nullptr) probably does the trick and avoids yet another bool member.

Reply:
It's not the same thing. This boolean checks whether the Gzip header was read after seeking. If it was the `m_streamBytesCount` would be accurate. on the other hand if there has been a `seek` inside the stream, it won't be accurate.